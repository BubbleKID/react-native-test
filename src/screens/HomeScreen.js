import React from 'react';
import {
  Platform, StatusBar, StyleSheet, View,
} from 'react-native';
import axios from 'axios';
import PostList from '../components/PostList';

export default class HomeScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = ({
      postData: [],
    });
  }

  componentDidMount() {
    axios.get('https://staging.allfin.com/wordpress/wp-json/wp/v2/posts?page=1&per_page=10')
      .then((response) => {
        this.setState({ postData: response.data });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    const { postData } = this.state;
    return (
      <View style={styles.container}>
        {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
        <PostList {...this.props} postData={postData} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#efeff4',
  },
});
