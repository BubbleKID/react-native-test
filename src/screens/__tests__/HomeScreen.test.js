import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../HomeScreen';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import { shallow } from 'enzyme';

it('renders correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('axios fetch data correctly', async () => {
  const spy = jest.spyOn(HomeScreen.prototype, 'componentDidMount');
  const mock = new MockAdapter(axios);
  mock.onGet('https://staging.allfin.com/wordpress/wp-json/wp/v2/posts?page=1&per_page=10').reply(200,
    {id: 1, name: 'John Smith'}
  );
  const wrapper = shallow(<HomeScreen />);
  expect(spy).toHaveBeenCalled();
});

it('axios fetch data error', async () => {
  const spy = jest.spyOn(HomeScreen.prototype, 'componentDidMount');
  const mock = new MockAdapter(axios);
  mock.onGet('https://staging.allfin.com/wordpress/wp-json/wp/v2/posts?page=1&per_page=10').networkError();
  const wrapper = shallow(<HomeScreen />);
  console.log = jest.fn();
  expect(spy).toHaveBeenCalled();
});


