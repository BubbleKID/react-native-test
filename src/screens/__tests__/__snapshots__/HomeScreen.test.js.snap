// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`axios fetch data correctly 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#efeff4",
      "flex": 1,
    }
  }
/>
`;
